### PR TITLE
Database

### PR DESCRIPTION
**Pull request by fungiboletus**
This pull-request adds the database schema and the corresponding Hasura configuration.

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+src/db/database.ts
```
Comment (fungiboletus) [inline .eslintignore:1]: This file is auto-generated and doesn't follow eslint rules.

**Comment by fungiboletus** [comment]
@AleenaThomas-gh Can you review the pull-request ? A lot of files are generated, so you don't need to check everything carefully but maybe reading quickly once `up.sql` would be nice, if you spot stupid mistakes or errors.

File: hasura/metadata/databases/default/tables/simpipe_runs.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  using:
+    foreign_key_constraint_on:
+      column: run_id
+      table:
+        name: steps
+        schema: simpipe
+select_permissions:
+- permission:
+    columns:
+    - simulation_id
```
Comment (AleenaThomas-gh) [inline hasura/metadata/databases/default/tables/simpipe_runs.yaml:28]: Isn't there overlapping columns between simpipe_runs and simpipe_steps? (ie, created, status, ended,started)

**Review by AleenaThomas-gh — APPROVED** [review]

File: hasura/metadata/databases/default/tables/simpipe_runs.yaml (hunk 2 above)
Comment (fungiboletus) [reply hasura/metadata/databases/default/tables/simpipe_runs.yaml:28]: It's the same columns and concept, but it shouldn't be the same data. The runs columns could probably be computed from the steps columns though, but that will make things a bit more complex. What do you think ?

Comment (AleenaThomas-gh) [reply hasura/metadata/databases/default/tables/simpipe_runs.yaml:28]: Yeah, on second thoughts it seems better to keep the timestamp columns separate as it is now. Maybe the run_started and run_ended would be a slightly longer interval than step1's and step_last's corresponding values.

**Pull request merged**